AnarchyExploitFixesFolia/src/main/java/me/xginko/aef/modules/illegals/items/IllegalItemModule.java

```diff
@@ -81,8 +85,10 @@ public IllegalItemModule(String configPath, AEFPermission bypassPermission) {
                 @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
                 private void onInventoryOpen(InventoryOpenEvent event) {
                     if (!CachingPermTool.hasPermission(bypassPermission, event.getPlayer())) {
-                        for (ItemStack invItem : event.getInventory()) {
-                            handleItem(invItem, legalityOf(invItem));
+                        if (!guiPluginsSupported || (guiPluginsSupported && event.getInventory().getLocation() != null)) { // Check if the inventory is connected to a location in the game. If it is not, its very likely created by custom gui plugin
```

Comment:
That second or-check is redundant, if guiPluginsSupported is true, the first part of the or check will always be false, meaning we only need to check if the location is not null

Reply:
-> if (!guiPluginsSupported || event.getInventory().getLocation() != null)